### PR TITLE
Add acceptance tests for how provider handles `impersonate_service_account` argument

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR adds acceptance tests for usage of `impersonate_service_account` that demonstrate:

- how the provider behaves when provider configuration arguments come from different sources ( config vs ENVs)
    - There is only a single ENV used for this argument, versus other arguments
- schema-level validation that's in place, e.g. handling of empty strings
- use cases: how does this argument impact the providers behaviour in plan/apply

**Note: this PR does not include testing usage of impersonate_service_account when using PF-implemented resource/data source as doing this in the context of Firebase (the only service with PF-implemented stuff) is very complex.**

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by SarahFrench — COMMENTED** [review]
Some comments to assist in review

**Comment by SarahFrench** [comment]
Some TeamCity builds running these tests:

- SDK : [TestAccSdkProvider_impersonate_service_account](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_PROVIDER/238923?buildTab=overview&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)
- PF : [TestAccFwProvider_impersonate_service_account](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_FWPROVIDER/238924?buildTab=overview&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)

**Comment by SarahFrench** [comment]
Marking this as ready for review - happy to save merging for after the rewrite is released (will reduce the diffs in this PR) but would be nice to get feedback on the tests in the meantime

I've [self-reviewed a little here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11641#pullrequestreview-2322676601) and have triggered the tests in TeamCity [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11641#issuecomment-2369005039).

**Review by c2thorn — APPROVED** [review]
LGTM overall. Can't think of more cases to cover with `impersonate_service_account`

File: mmv1/third_party/terraform/provider/provider_impersonate_service_account_test.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+provider "google" {
+	impersonate_service_account = "%{impersonate_service_account}"
+}
+
+data "google_provider_config_sdk" "default" {}
+
+output "impersonate_service_account" {
+  value = data.google_provider_config_sdk.default.impersonate_service_account
+  sensitive = true
+}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/provider/provider_impersonate_service_account_test.go:152]:
```suggestion
```

File: mmv1/third_party/terraform/provider/provider_impersonate_service_account_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+// testAccSdkProvider_impersonate_service_account_inEnvsOnly allows testing when the impersonate_service_account argument
+// is only supplied via ENVs
+func testAccSdkProvider_impersonate_service_account_inEnvsOnly(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+data "google_provider_config_sdk" "default" {}
+
+output "impersonate_service_account" {
+  value = data.google_provider_config_sdk.default.impersonate_service_account
+  sensitive = true
+}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/provider/provider_impersonate_service_account_test.go:165]:
```suggestion
```

File: mmv1/third_party/terraform/fwprovider/framework_provider_impersonate_service_account_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+// testAccFwProvider_impersonate_service_account_inEnvsOnly allows testing when the impersonate_service_account argument
+// is only supplied via ENVs
+func testAccFwProvider_impersonate_service_account_inEnvsOnly(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+data "google_provider_config_plugin_framework" "default" {}
+
+output "impersonate_service_account" {
+  value = data.google_provider_config_plugin_framework.default.impersonate_service_account
+  sensitive = true
+}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwprovider/framework_provider_impersonate_service_account_test.go:145]:
```suggestion
```

File: mmv1/third_party/terraform/fwprovider/framework_provider_impersonate_service_account_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+provider "google" {
+	impersonate_service_account = "%{impersonate_service_account}"
+}
+
+data "google_provider_config_plugin_framework" "default" {}
+
+output "impersonate_service_account" {
+  value = data.google_provider_config_plugin_framework.default.impersonate_service_account
+  sensitive = true
+}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwprovider/framework_provider_impersonate_service_account_test.go:132]:
```suggestion
```

**Review by SarahFrench — COMMENTED** [review]
Thanks for the review!

Before merging I'm going to remove these outputs that are unused.

**Comment by ScottSuarez** [comment]
Hi Sarah, this test is failing in teamcity at 50% frequency 

https://hashicorp.teamcity.com/test/-2975064263368697375?currentProjectId=TerraformProviders_GoogleCloud_GOOGLE_BETA_NIGHTLYTESTS&expandTestHistoryChartSection=true

I suspect because the actual SA's we are attempting to impersonate do not exist 
https://github.com/hashicorp/terraform-provider-google/issues/19741

**Comment by SarahFrench** [comment]
Thanks for letting me know! I'll split the test into two steps and that should ensure the service account is present.

**Pull request merged**